samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.scaleout;
+
+import com.microsoft.bot.builder.ActivityHandler;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.dialogs.Dialog;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Represents a bot that processes incoming Activities.
+ * For each user interaction, an instance of this class is created and the OnTurnAsync method is called.
+ */
+public class ScaleoutBot <T extends Dialog> extends ActivityHandler {
+
+    private final Store store;
+    private final Dialog dialog;
+
+    /**
+     * Initializes a new instance of the {@link ScaleoutBot} class.
+     * @param withStore The store we will be using.
+     * @param withDialog The root dialog to run.
+     */
+    public ScaleoutBot(Store withStore, T withDialog) {
+        if (withStore == null) {
+            throw new IllegalArgumentException("withStore can't be null");
+        }
+        store = withStore;
+
+        if (withDialog == null) {
+            throw new IllegalArgumentException("withDialog can't be null");
+        }
+        dialog = withDialog;
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
```

Comment:
Can you add the [javadocs](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/42.scaleout/Bots/ScaleoutBot.cs#L35)?

Reply:
Yes, I added as a draft as I had some missing changes that I could apply right now